File: tournaments/ballsDistribution/ballsDistribution.cpp
int ballsDistribution(int colors, int ballsPerColor, int boxSize) {

  int currentBox = 0;
  int capacity = boxSize;
  int result = 0;

  for (int i = 0; i < colors; i++) {
    int startBox = currentBox;
    for (int j = 0; j < ballsPerColor; j++) {
      capacity--;
      if (capacity == 0) {
        currentBox++;
        capacity = boxSize;
      }
    }
    if ((startBox < currentBox && capacity < boxSize) ||
         startBox + 1 < currentBox) {
      result++;
    }
  }

  return result;
}

File: tournaments/ballsDistribution/ballsDistribution.js
function ballsDistribution(colors, ballsPerColor, boxSize) {
    let result = 0;
    let a = 0;
    let b = boxSize;

    for (let i = 0; colors > i; ++i) {
        let c = a;
        for (let j = 0; ballsPerColor > j; ++j) {
            --b;
            if (!b) {
                b = boxSize;
                ++a;
            }
        }
        if (c < a && (boxSize > b || c + 1 < a)) ++result;
    }

    return result;
}
